### PR TITLE
Integrate `workflow.pacta` analysis in dashboard generation

### PR DESCRIPTION
**Pull request by AlexAxthelm**

### CODE REVIEW

**Comment by jdhoffa** [comment]
> [PR Preview Action](https://github.com/rossjrw/pr-preview-action) v1.4.8
> 🚀 Deployed preview to https://RMI-PACTA.github.io/workflow.pacta.dashboard/pr-preview/pr-49/
> on branch [`gh-pages`](https://github.com/RMI-PACTA/workflow.pacta.dashboard/tree/gh-pages) at 2024-12-13 11:46 UTC

This is so awesome!


Just noting that for:
[default_2023Q4](https://rmi-pacta.github.io/workflow.pacta.dashboard/pr-preview/pr-49/dashboard-build-default_2023Q4/index.html)
and
[full_params_2023Q4](https://rmi-pacta.github.io/workflow.pacta.dashboard/pr-preview/pr-49/dashboard-build-full_params_2023Q4/index.html)

The portfolio-level technology mix plots don't appear:
Screenshot 2024-12-13 at 12 56 33

File: tests/config/default_2022Q4.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "holdingsDate": "2022Q4",
+  "pactaDataURL": "https://pactadatadev.blob.core.windows.net/pacta-data-webapp/2022Q4/2022Q4_20240426T113151Z",
+  "benchmarksURL": "https://pactadatadev.blob.core.windows.net/benchmarks-webapp/2022Q4/2022Q4_20240529T002407Z",
+  "parameters": {
+    "portfolio": {
+      "holdingsDate": "2023-12-31",
```
Comment (jdhoffa) [inline tests/config/default_2022Q4.json:7]:
```suggestion
      "holdingsDate": "2022-12-31",
```
Relates to https://github.com/RMI-PACTA/workflow.pacta/pull/105

File: tests/config/full_params_2022Q4.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+  "holdingsDate": "2022Q4",
+  "pactaDataURL": "https://pactadatadev.blob.core.windows.net/pacta-data-webapp/2022Q4/2022Q4_20240426T113151Z",
+  "benchmarksURL": "https://pactadatadev.blob.core.windows.net/benchmarks-webapp/2022Q4/2022Q4_20240529T002407Z",
+  "parameters": {
+    "portfolio": {
+      "holdingsDate": "2023-12-31",
```
Comment (jdhoffa) [inline tests/config/full_params_2022Q4.json:7]:
```suggestion
      "holdingsDate": "2022-12-31",
```

Relates to https://github.com/RMI-PACTA/workflow.pacta/pull/105

File: tests/config/full_params_2022Q4.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      ],
+      "sectorList": [
+        "Power",
+        "Automotive",
+        "Coal",
+        "Steel",
+        "Aviation",
+        "Cement"
+      ],
+      "startYear": 2023,
```
Comment (jdhoffa) [inline tests/config/full_params_2022Q4.json:38]:
```suggestion
      "startYear": 2022,
```

File: tests/config/full_params_2023Q4.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      ],
+      "pactaSectorsNotAnalysed": [
+        "Steel",
+        "Aviation",
+        "Cement"
+      ],
+      "portfolioAllocationMethod": "portfolio_weight",
+      "scenarioGeography": "Global",
+      "scenarioOther": "GECO2022_1.5C",
+      "selectScenario": "WEO2022_NZE_2050",
```
Comment (jdhoffa) [inline tests/config/full_params_2023Q4.json:68]:
```suggestion
      "selectScenario": "WEO2023_NZE_2050",
```

File: tests/config/full_params_2023Q4.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        "Electric Arc Furnace"
+      ],
+      "pactaSectorsNotAnalysed": [
+        "Steel",
+        "Aviation",
+        "Cement"
+      ],
+      "portfolioAllocationMethod": "portfolio_weight",
+      "scenarioGeography": "Global",
+      "scenarioOther": "GECO2022_1.5C",
```
Comment (jdhoffa) [inline tests/config/full_params_2023Q4.json:67]:
```suggestion
      "scenarioOther": "WEO2023_NZE_2050",
```

File: tests/config/default_2023Q4.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  "benchmarksURL": "https://pactadatadev.blob.core.windows.net/benchmarks-webapp/2023Q4/2023Q4_20240529T002355Z",
+  "parameters": {
+    "portfolio": {
+      "holdingsDate": "2023-12-31",
+      "files": [
+        "default_portfolio.csv"
+      ],
+      "name": "Default Portfolio"
+    },
+    "inherit": "GENERAL_2023Q4"
```
Comment (jdhoffa) [inline tests/config/default_2023Q4.json:13]: Where is `GENERAL_2023Q4` defined?

**Review by jdhoffa — CHANGES_REQUESTED** [review]

File: tests/config/default_2023Q4.json (hunk 6 above)
Comment (jdhoffa) [reply tests/config/default_2023Q4.json:13]: I guess that will depend on https://github.com/RMI-PACTA/workflow.pacta.dashboard/pull/47 being merged?

**Comment by jdhoffa** [comment]
> Interestingly, this seems to also be the case for our test CI on `worklow.pacta.webapp`: See: [RMI-PACTA/workflow.pacta.webapp#27](https://github.com/RMI-PACTA/workflow.pacta.webapp/pull/27)

@AlexAxthelm  This leads me to believe that this is an artifact of the test data (or something else) and not directly related to this PR, so not-blocking. 

I will look into the data side, and you can keep on with this 😊

File: tests/config/default_2023Q4.json (hunk 6 above)
Comment (AlexAxthelm) [reply tests/config/default_2023Q4.json:13]: This is targeting #47, not main.

Comment (jdhoffa) [reply tests/config/default_2023Q4.json:13]: Wooops! 😅 good flag

**Comment by jdhoffa** [comment]
> I will look into the data side, and you can keep on with this 😊

Reproduced locally.

**Comment by jdhoffa** [comment]
> > I will look into the data side, and you can keep on with this 😊
> 
> Reproduced locally.

The bug if that the selected `selectscenario` values for `2023Q4` point to `2022` scenarios:
``` r
Browse[1]> select_scenario_other
[1] "GECO2022_1.5C"
Browse[1]> select_scenario
[1] "WEO2022_NZE_2050"
```

I will review all configs across `workflow.pacta`, `workflow.pacta.report`, `workflow.pacta.webapp` and `workflow.pacta.dashboard` and fix that.

**Comment by AlexAxthelm** [comment]
Merging this PR into #47

**Pull request merged**